### PR TITLE
Basic local test support

### PR DESCRIPTION
**Pull request by kasparsd**
Introduce a Docker based dev setup for running the PHP unit tests:

- `npm run docker-test` to run the unit tests.
- `npm run docker-lint` to run everything _but_ the unit tests.
- `npm run docker-cli` to support any command.

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -10,22 +10,19 @@
...
 	</php>
 	<testsuites>
-		<testsuite>
+		<testsuite name="All Tests">
 			<directory suffix=".php">tests</directory>
 		</testsuite>
 	</testsuites>
 	<filter>
 		<whitelist processUncoveredFilesFromWhitelist="false">
-			<directory suffix=".php">.</directory>
```
Comment (kasparsd) [inline phpunit.xml:19]: This was causing 10 second delay before tests while it parsed all files in the excluded directories.

File: tests/docker/wp-dev-lib/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+ARG WP_PHP_VERSION=7.4
```
Comment (kasparsd) [inline tests/docker/wp-dev-lib/Dockerfile:1]: We're keeping it here for now and then moving to wp-dev-lib repo and eventually the Docker Hub.

**Comment by kasparsd** [comment]
@paulschreiber Could you please try this for running the unit tests locally? All you need is a Docker engine running on the host.

**Comment by paulschreiber** [comment]
The Docker image is using `htto` and not `https` for updates:
```
Get:1 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
Get:2 http://deb.debian.org/debian buster InRelease [121 kB]
Get:3 http://security.debian.org/debian-security buster/updates/main amd64 Packages [201 kB]
Get:4 http://deb.debian.org/debian buster-updates InRelease [49.3 kB]
Get:5 http://deb.debian.org/debian buster/main amd64 Packages [7905 kB]
Get:6 http://deb.debian.org/debian buster-updates/main amd64 Packages [7380 B]
```

**Comment by paulschreiber** [comment]
running `npm run docker-test` modified readme.md with an updated `Build status`

File: tests/docker/wp-dev-lib/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ARG WP_SVN_URL=https://develop.svn.wordpress.org/trunk/
+
+ENV DEBIAN_FRONTEND noninteractive
+
+# Development tooling dependencies.
+RUN apt-get update \
+	&& apt-get install --yes --no-install-recommends \
+		bash less subversion default-mysql-server default-mysql-client libxml2-utils rsync git zip unzip \
+		nodejs npm curl \
+	&& npm install --global npm@latest \
```
Comment (kasparsd) [inline tests/docker/wp-dev-lib/Dockerfile:14]: @paulschreiber This is a fix for that "Warning about old npm:" message. Here we bump it up.

**Comment by kasparsd** [comment]
> The Docker image is using htto and not https for updates:

Yeah, Debian does that for these reasons https://whydoesaptnotusehttps.com -- nothing we can do about this really.

> running npm run docker-test modified readme.md with an updated Build status

Thanks for pointing this out. That the `wp-dev-lib` doing that here:

https://github.com/xwp/wp-dev-lib/blob/e9f2a931eff50e1c46209de0b86d720a65966144/scripts/pre-commit#L65-L70

Looks like we need to remove it from the test runner by setting the environment variable `YNC_README_MD=0`.

**Pull request merged**